providers/default.rb

```diff
@@ -637,7 +637,8 @@ def execute_install_script(install_script)
                     '/etc/init.d/chef-client start'
                   end
       unless node['chef_client']['chef_license'].nil?
-        start_cmd = "env CHEF_LICENSE=\"#{node['chef_client']['chef_license']}\" " + start_cmd
+        license_acceptance = shell_out("#{chef_install_dir}/bin/chef-apply -e 'exit 0'", timeout: 60, environment: { 'CHEF_LICENSE' => "#{node['chef_client']['chef_license']}" })
+        Chef::Log.warn 'Something went wrong while accepting the license.' if license_acceptance.exitstatus != 0
```

Comment:
If we're going to have nesting this deep in only this place, is it worth disabling the rubocop check that's failing for this line?

```
# rubocop:disable Metrics/BlockNesting
Chef::Log.warn 'Something went wrong while accepting the license.' if license_acceptance.exitstatus != 0
# rubocop:enable Metrics/BlockNesting
```

or this if it's something we want to revisit later (not sure what that'll do in terms of the build, though):
```
# rubocop:todo Metrics/BlockNesting
```

Reply:
It would probably be more in the spirit of the warning to extract a function like `accept_license!` and move attributes check and shell_out call there.